packages/@orbit/jsonapi/src/lib/transform-requests.ts

```diff
@@ -57,13 +42,18 @@ export const TransformRequestProcessors = {
   },
 
   replaceRecord(source: JSONAPISource, request) {
+    const { serializer } = source;
     const record = request.record;
     const { type, id } = record;
-    const requestDoc: JSONAPIDocument = source.serializer.serializeDocument(record);
+    const requestDoc: JSONAPIDocument = serializer.serializeDocument(record);
     const settings = buildFetchSettings(request.options, { method: 'PATCH', json: requestDoc });
 
     return source.fetch(source.resourceURL(type, id), settings)
-      .then(() => []);
+      .then((raw: JSONAPIDocument) => {
+        if (raw) {
+          return handleChanges(record, serializer.deserializeDocument(raw, record));
+        }
```

Comment:
I believe this needs to always return an array:

```suggestion
        if (raw) {
          return handleChanges(record, serializer.deserializeDocument(raw, record));
        } else {
          return [];
        }
```